### PR TITLE
Add origin types kubectl scheme

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/openshift/origin/issues/17793

cc @deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
See travis for the revert commit format

File: cmd/oc/oc.go [hunk 1]
```diff
@@ -31,6 +46,20 @@ func main() {
 		runtime.GOMAXPROCS(runtime.NumCPU())
 	}
 
+	// add Origin types to the kubectl scheme
+	apps.Install(scheme.GroupFactoryRegistry, scheme.Registry, scheme.Scheme)
```
Comment (deads2k) [inline cmd/oc/oc.go]: Please make an `InstallAll` method in `pkg/api/install`

**Comment by deads2k** [comment]
right idea.  Nit on commit name and where the big-button install lives.

**Comment by juanvallejo** [comment]
@deads2k thanks, added an InstallAll method

File: cmd/oc/oc.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/openshift/origin/pkg/oc/cli"
 
 	// install all APIs
-	_ "github.com/openshift/origin/pkg/api/install"
+	api "github.com/openshift/origin/pkg/api/install"
```
Comment (deads2k) [inline cmd/oc/oc.go]: apinstall

Comment (deads2k) [reply cmd/oc/oc.go]: apiinstall

File: pkg/api/install/install.go [hunk 3]
```diff
@@ -592,3 +594,19 @@ func init() {
 		return false, nil
 	})
 }
+
+func InstallAll() {
```
Comment (deads2k) [inline pkg/api/install/install.go]: This function should take a scheme as an argument

**Comment by juanvallejo** [comment]
@deads2k thanks, review comments addressed

File: pkg/api/install/install.go [hunk 4]
```diff
@@ -592,3 +596,19 @@ func init() {
 		return false, nil
 	})
 }
+
+func InstallAll(scheme *runtime.Scheme, groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *registered.APIRegistrationManager) {
+	// add Origin types to the kubectl scheme
```
Comment (deads2k) [inline pkg/api/install/install.go]: nit: comment is now wrong

**Comment by deads2k** [comment]
nit, squash last two commits, then tag at will.

**Comment by deads2k** [comment]
@juanvallejo try against master.  My guess is that its this pull.

**Comment by juanvallejo** [comment]
@deads2k looks like even with the `InstallAll` method being called, using kubectl scheme instead of legacyscheme in apply causes an `IsNotRegisteredError` here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/apply.go#L612 which falls back to patching an object using a non-strategic merge patch.

This is the working strategic three-way merge patch that is generated when using `legacyscheme`:
http://pastebin.test.redhat.com/541776

This is the broken patch that is generated when switching to kubectl/scheme (with the changes from this PR):
http://pastebin.test.redhat.com/541779

**Comment by deads2k** [comment]
> @deads2k looks like even with the InstallAll method being called, using kubectl scheme instead of legacyscheme in apply causes an IsNotRegisteredError here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/apply.go#L612 which falls back to patching an object using a non-strategic merge patch.

Probably need to figure out which object and why.  That doesn't match my expectation.

**Comment by juanvallejo** [comment]
> Probably need to figure out which object and why. That doesn't match my expectation.

It's an ImageStream: http://pastebin.test.redhat.com/541789
being patched with: http://pastebin.test.redhat.com/541786

Will look into why

**Comment by juanvallejo** [comment]
closing in favor of https://github.com/openshift/origin/pull/17885

**Pull request closed without merging**